### PR TITLE
add: button component

### PR DESCRIPTION
**Pull request by shilpe26**
Made Button Component with features:-
🔹Simple hover Buttons
🔹Link Button
🔹Icon Button
🔹Floating Action Button
Button

### CODE REVIEW

File: css/component_styles/button.css [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    border-color: var(--primary-hover);
+}
+
+.btn-primary:hover {
+    background-color: var(--primary-hover);
+}
+
+.space-ht {
+    margin-bottom: 1rem;
+}
```
Comment (thegeorgenikhil) [inline css/component_styles/button.css]: `mb-1` would be a better naming for this utility class

File: css/component_styles/button.css [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+.button-icon:hover {
+    background-color: var(--primary-hover);
+    color: black;
+}
+
+span {
+    padding: 1rem;
+}
+
```
Comment (thegeorgenikhil) [inline css/component_styles/button.css]: refrain from using styles directly on elements

**Review by thegeorgenikhil — CHANGES_REQUESTED** [review]

File: components/button.html [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <!-- Icon link -->
+    <link rel="icon" href="../assets/icon.png">
+    <!-- css style links -->
+    <link rel="stylesheet" href="../css/base_styles/reset.css">
+    <link rel="stylesheet" href="../css/base_styles/home.css">
+    <link rel="stylesheet" href="../css/component_styles/button.css">
+    <link rel="stylesheet" href="../css/component_styles/alert.css">
+    <link rel="stylesheet" href="../css/base_styles/introduction.css">
+    <link rel="stylesheet" href="../css/base_styles/media.css">
```
Comment (junaidshaikh-js) [inline components/button.html]: You can import all your CSS into one file and attach a single file to your HTML files.

File: components/button.html [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    <title>Button</title>
+
+    <!-- Font Awesome link -->
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.13.0/css/all.css"
+        integrity="sha384-Bfad6CLCknfcloXFOyFnlgtENryhrpZCe29RTifKEixXQZ38WheV+i/6YWSzkz3V" crossorigin="anonymous" />
+</head>
+
+<body>
+    <!-- navbar-section -->
+    <nav class="navigation">
```
Comment (junaidshaikh-js) [inline components/button.html:25]: `nav` element can be nested inside `header` element.

**Review by anuj-kumary — CHANGES_REQUESTED** [review]
One suggestion give some more padding to the icon button

File: components/button.html [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        <div class="nav-brand"><a class="link" href="/index.html">Rebalance-UI</a></div>
+        <ul class="non-bullet">
+            <li class="list-tem-inline"><a class="link link-active" href="/index.html">Home</a></li>
+            <li class="list-tem-inline"><a class="link" href="./avatar.html">Docs</a></li>
+        </ul>
+    </nav>
+
+    <!-- side-list bar -->
+    <div class="aside-container">
+        <aside class="aside-nav">
```
Comment (junaidshaikh-js) [inline components/button.html:36]: use `main` element for all your main content and nest `aside` inside `main` element.

File: components/button.html [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+                </li>
+                <li class="navbar-vertical-item">
+                    <a class="aside-link" href="./toast.html">Toast</a>
+                </li>
+            </ul>
+        </aside>
+
+
+        <!-- main-section -->
+        <main class="avatar-main">
```
Comment (junaidshaikh-js) [inline components/button.html:85]: Great, you used `main` element but `aside` can be nested inside `main`

File: components/button.html [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            <h1 class="avatar-heading">Button</h1>
+            <p class="avatar-text">
+                The <code>button</code> HTML element represents a clickable button, used to submit forms or anywhere in
+                a document
+                for accessible, standard button functionality.
+            </p>
+            <div class="main-container">
+                <h1>Usage</h1>
```
Comment (junaidshaikh-js) [inline components/button.html]: There should be only one `h1` heading per page, use [heading hierarchy](https://www.w3.org/WAI/tutorials/page-structure/headings/)

**Review by MurtazaBagwala07 — APPROVED** [review]

File: css/component_styles/button.css [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    background-color: var(--info-color);
+    border-color: var(--info-hover);
+}
+
+.btn-info:hover {
+    background-color: var(--info-hover);
+    color: black;
+}
+
+#avatar-iframe5 {
```
Comment (junaidshaikh-js) [inline css/component_styles/button.css]: IDs are not recommended to use for styling, use some other selector.

**Comment by shilpe26** [comment]
Done all the changes as suggested👍🏻

**Pull request merged**